columns/Column.js

```diff
@@ -192,13 +200,19 @@ export class Column {
 
         const {renderer, elementRenderer} = this;
         if (renderer) {
-            ret.cellRenderer = (params) => {
-                const metaData = {colId: params.column.colId};
-                return renderer(params.value, params.data, metaData);
+            ret.cellRenderer = (agParams) => {
+                return renderer(agParams.value, {record: agParams.data, column: this, agParams});
             };
         } else if (elementRenderer) {
+            // eslint-disable-next-line consistent-this
+            const column = this;
             ret.cellRendererFramework = class extends Component {
-                render() {return elementRenderer(this.props)}
+                render() {
+                    const agParams = this.props,
+                        {value, data: record} = agParams;
+
+                    return elementRenderer({value, record, agParams, column});
+                }
                 refresh() {return false}
```

Comment:
If we are going to mess with these, elementRenderer and renderer callbacks really should end up with the same exact signature.  I think I would prefer renderer(val, Object) but would also be ok with renderer(Object).  In any case, we need to choose one and use the JSDoc callback notation to clearly document it.

Reply:
The only problem with renderer(val, Object) is that you then can't just provide an elemFactory for the elementRenderer. Maybe that's fine and we want to make it clear that the elementRenderer callback and elemFactory are not the same thing? 

I don't think we want to change the cellRenderer signature to take an object, @amcclain made a good point in #575 about that

Reply:
OK -- totally fine with going that way.  I also think its totally fine that you can't provide a raw elemFactory to the renderer -- as you saw from my other comment want to move away from any possible conflation of the two concepts....